lib/truemail.rb

```diff
@@ -1,5 +1,18 @@
 require 'truemail/version'
+require 'truemail/exceptions'
+require 'truemail/error'
+require 'truemail/configuration'
 
 module Truemail
-  class Error < StandardError; end
+  class << self
+    attr_accessor :configuration
+
+    def configure
```

Comment:
def configuration
 @configuration ||= begin do
     configuration = Configuration.new
     raise ConfigurationError, Error::CONFIGURATION_INCOMPLETE unless configuration.complete?
     configuration
 end
end

Reply:
This code will never work. My refactor:

```ruby
module Truemail
  def self.configuration
    @configuration ||= begin
      return unless block_given?
      configuration = Configuration.new
      yield(configuration)
      raise ConfigurationError, ConfigurationError::INCOMPLETE_CONFIG unless configuration.complete?
      configuration
    end
  end

  def self.configure(&block)
    configuration(&block)
  end
end
```